### DISCUSSION TITLE
Feature Request: the option to publish a sketch version

### DISCUSSION THREAD

**Discussion by Gabrz** [processing/p5.js-web-editor#2666] [General]
I would like to see an option to "publish" a sketch.

I've shared my sketch (on social media) using the HTML link. But I would like to continue working on the sketch.

It would be nice if the link would go to a "published" / working sketch, and I can keep on making changes without disrupting the shared/published sketch. And when I'm done editing I can publish it, and the shared sketch is updated to the new version.

**Comment by welcome** [comment]
Welcome! 👋 Thanks for opening your first issue here! And to ensure the community is able to respond to your issue, be sure to follow the issue template if you haven't already.

**Comment by lindapaiste** [comment]
~~I'm thinking that what makes the most sense here is some action that would take the entire current sketch and copy it to a new sketch.  Call it "duplicate", "fork", "clone", "save as...", or whatever.~~

I didn't realize that there is already a "duplicate" option! 😝 So you can work on a duplicated copy while keeping the original intact.  What's missing is the ability to then go back and push changes from the cloned duplicated into your published original, as you said here:
> And when I'm done editing I can publish it, and the shared sketch is updated to the new version.

Interesting idea.  I'm not sure of the logistics of it.
